src/tags/render.js

```diff
@@ -56,7 +56,7 @@ module.exports = function (fractal) {
                 let innerContext = entity.isComponent ? entity.variants().default().getContext() : entity.getContext();
 
                 if (token.contextStack !== undefined) {
-                    _.assign(innerContext, Twig.expression.parse.apply(this, [token.contextStack, context]));
+                    _.merge(innerContext, Twig.expression.parse.apply(this, [token.contextStack, context]));
```

Comment:
this is usually fine, but `_.merge` does not work very well with merging arrays.

For example, if the rendered component has an array in context:
```yaml
context:
  arr:
    - first
    - second
```
and you want to override the array to have just one item:
`{% render '@comp' with { arr: ['newfirst'] } %}`, the lodash merge actually keeps the second array item and returns `['newfirst', 'second']`.

Other Fractal [adapters](https://github.com/frctl/nunjucks/blob/develop/src/extensions/render.js#L37) use Fractals own `defaultsDeep` function from utils that behaves like `_.merge` for objects, but like `_.assign` for arrays. I'd reckon we're better of using that here as well to be more compatible with other adapters.
```suggestion
                    innerContext = utils.defaultsDeep(Twig.expression.parse.apply(this, [token.contextStack, context]), innerContext);
```